src/components/GameItem.js

```diff
@@ -4,14 +4,25 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      clicked: false 
+    }
+  }
+  
   static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
   onItemClicked = () => {
-    // Fill this in for Wave 2!
-  }
+    this.setState({
+      clicked: true 
+    })
+
+    this.props.onItemClickedCallback(this.props.type)
```

Comment:
A check for clicked state could be added here to fix the score but. Then this line would also need to be moved before the state is updated.